projects/bootloader/src/drivers/flash/flash_driver.h

```diff
@@ -9,12 +9,59 @@
  *
  */
 
-// --- includes --------------------------------------------------------------------------------------------------------
+#ifndef FLASH_DRIVER_H
+#define FLASH_DRIVER_H
 
-// --- defines ---------------------------------------------------------------------------------------------------------
+// --- includes --------------------------------------------------------------------------------------------------------
+#include <stdint.h>
+#include <stdbool.h>
 
 // --- typedefs --------------------------------------------------------------------------------------------------------
+typedef struct
+{
+    uint32_t start_address;
+    uint32_t size;
+} flash_sector;
+
+// --- static variable definitions -------------------------------------------------------------------------------------
+// STM32F401RE supports flash erase in sectors.
+// clang-format off
+/*
+    |sec idx  |size kB|start addr |
+    |Sector 0 | 16kB  |0x08000000| Bootloader memory
+    |Sector 1 | 16kB  |0x08004000| Bootloader memory
+    |Sector 2 | 16kB  |0x08008000| App memory primary
+    |Sector 3 | 16kB  |0x0800C000| App memory primary
+    |Sector 4 | 64kB  |0x08010000| App memory primary
+    |Sector 5 | 128kB |0x08020000| App memory primary - total 224kB available for app primary
+    |Sector 6 | 128kB |0x08040000| App secondary
+    |Sector 7 | 128kB |0x08060000| App secondary - end address 0x08077FFF - app primary size == app secondary size
+*/
 
-// --- enums -----------------------------------------------------------------------------------------------------------
+// TODO: GPA: Need to properly split flash to allow application have a dedicated flash space for persistent storage.
+static const flash_sector flash_sectors[] = {
+    {0x08000000, 16 * 1024},
```

Comment:
Using defines to name these sections on par with the reference manual would be nice.

Reply:
Done